src_python/org_agg.py

```diff
@@ -0,0 +1,238 @@
+            return False
+
+    def run(self):
+
+        while not self.stop_flag:
+
+            read_sockets, write_sockets, exception_sockets = select.select(self.sockets_list, [], self.sockets_list)
+
+            for notified_socket in read_sockets:
+
+                if notified_socket == self.aggregator_socket:
+
+                    client_socket, client_address = self.aggregator_socket.accept()
+                    self.sockets_list.append(client_socket)
+                    self.clients[client_socket] = client_address
+                    self.client_ids[client_socket] = "new"
+
+                    logging.info('Accepted new connection at %s:%s',*client_address)
+
+                    self.send_model(client_socket)
+
+                else:
+                    
+                    message = self.receive(notified_socket)
+
+                    if message is False:
+                        self.sockets_list.remove(notified_socket)
+                        del self.clients[notified_socket]
+                        continue
+                    else:
+                        client_id = message['ORG_ID']
+                        weights = message['WEIGHTS']
+
+                    
+                        weights_path = "./weights/agg_a" + str(self.training_cycles) + ".npy"
+                        np.save(weights_path,weights)
+
+                        num_examples = message["NUM_EXAMPLES"]   #// commented
+                        self.client_ids[notified_socket] = client_id #// commented
+                    
+                    logging.info('Recieved model from client-%s at %s:%s',client_id, *self.clients[notified_socket])
+                    logging.info('Recieved model from client at %s:%s', *self.clients[notified_socket])
+                    self.update_model(weights,int(num_examples)) # commented
+
+            for notified_socket in exception_sockets:
+                self.sockets_list.remove(notified_socket)
+                del self.clients[notified_socket]
+
+if __name__ == "__main__":
+
+    from models.supervised import Model
+
+    arg_names = [
+        'path_nodes',
+        'path_edges',
+        'graph_id',
+        'partition_id',
+        'num_orgs',
+        'num_rounds',
+        'IP',
+        'PORT'
+        ]
+
+    args = dict(zip(arg_names, sys.argv[1:]))
+
+    logging.warning('####################################### New Training Session #######################################')
+    logging.info('aggregator started , graph ID %s, number of clients %s, number of rounds %s',args['graph_id'],args['num_orgs'],args['num_rounds'])
+
+    if 'IP' not in args.keys()  or args['IP'] == 'localhost':
+        args['IP'] = "localhost"
+
+    if 'PORT' not in args.keys():
+        args['PORT'] = 5000
+
+    path_nodes = args['path_nodes'] + args['graph_id'] + '_nodes_' + args['partition_id'] + ".csv"
+    nodes = pd.read_csv(path_nodes,index_col=0)
```

Comment:
From where do we read this file? We should have read this from JasmineGraph data directory. Not from the JasmineGraph server software's folder.

Reply:
`    std::string command = path + "python3 org_server.py " + graphID+ " " + utils.getJasmineGraphProperty("org.jasminegraph.fl_clients")
                                    + " " + utils.getJasmineGraphProperty("org.jasminegraph.fl.epochs") `
we input these values as parameter from jasminegraph worker

Reply:
These parameters (paths) can be only updated/changed by the jasminegraph conf file

Reply:
Done